lib/jellyfish/config_reader.ex

```diff
@@ -59,12 +59,31 @@ defmodule Jellyfish.ConfigReader do
         "false" ->
           false
 
-        _other ->
+        _other when is_nil(other_handler) ->
           raise "Bad #{env} environment variable value. Expected true or false, got: #{value}"
+
+        _other ->
+          other_handler.(value)
       end
     end
   end
 
+  def read_check_origin(env) do
+    parse_boolean(env, fn value ->
+      hosts = String.split(value, " ")
+
+      if Enum.all?(hosts, &String.contains?(&1, ".")) do
```

Comment:
As for these two additional cases, I wouldn't care about custom MFA because it will be only available for a user if he extends jellyfish by himself. For `:conn`, I can add another case that would match a string like this.